client/my-sites/add-ons/hooks/use-add-ons.ts

```diff
@@ -130,22 +130,25 @@ const useAddOns = ( siteId?: number ): ( AddOnMeta | null )[] => {
 		// get the list of supported features
 		const siteFeatures = getFeaturesBySiteId( state, siteId );
 		const filter = getBillingTransactionFilters( state, 'past' );
-		const filteredTransactions =
-			billingTransactions && filterTransactions( billingTransactions, filter, siteId );
-
 		const spaceUpgradesPurchased: number[] = [];
 
-		if ( filteredTransactions?.length ) {
-			for ( const transaction of filteredTransactions ) {
-				transaction.items?.length &&
-					spaceUpgradesPurchased.push(
-						...transaction.items
-							.filter( ( item ) => item.wpcom_product_slug === PRODUCT_1GB_SPACE )
-							.map( ( item ) => Number( item.licensed_quantity ) )
-					);
+		// It doesn't make sense to fetch purchase history and limit space upgrade add-ons if the user is
+		// creating a brand new site. For the time being, skip fetching transaction history in onboarding
+		// flows.
+		if ( billingTransactions && ! isInSignup ) {
+			const filteredTransactions = filterTransactions( billingTransactions, filter, siteId );
```

Comment:
If changing the block scope for `filteredTransactions`, then do you need to do the same for `filter` from above?

Reply:
Ah yes -- and I'm also now seeing that I missed one or two more issues. Thank you!

Reply:
Resolved in https://github.com/Automattic/wp-calypso/pull/81340/commits/2ea84520ba07526fab9b35e9958833ecf780f56f